assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -457,6 +494,189 @@ public GraderUser(User sakaiUser) {
         }
     }
 
+    @EntityCustomAction(action = "addTimeSheet", viewKey = EntityView.VIEW_NEW)
+    public BuildTimeSheetReturnMessage addTimeSheet(Map<String, Object> params) {
+
+        String userId = sessionManager.getCurrentSessionUserId();
+
+        if (StringUtils.isBlank(userId)) {
+          log.warn("You need to be logged in to add time sheet register");
+          return new BuildTimeSheetReturnMessage(false, 1, "ts.add.err.userId");
+        }
+        
+        User u;
+        try {
+            u = userDirectoryService.getUser(userId);
+        } catch (UserNotDefinedException e2) {
+            log.warn("You need to be logged in to add time sheet register");
+            return new BuildTimeSheetReturnMessage(false, 1, "ts.add.err.userId");
+        }
+
+        String assignmentId = (String)params.get("assignmentId");
+        if (StringUtils.isBlank(assignmentId)) {
+
+            log.warn("You need to supply the assignmentId and ref");
+            return new BuildTimeSheetReturnMessage(false, 1, "ts.add.err.assignmentId");
+        }
+
+        AssignmentSubmission as;
+        try {
+            as = assignmentService.getSubmission(assignmentId, u);
+        } catch (PermissionException e1) {
+            log.warn("You can't modify this sumbitter");
+            return new BuildTimeSheetReturnMessage(false, 1, "ts.add.err.permission");
+        }
+
+        if(as == null) {
+            String submitterId = u.getId();
+
+            try {
+                as = assignmentService.addSubmission(assignmentId, submitterId);
+                if (as != null ) {
+                    as.setSubmitted(true);
+                    as.setUserSubmission(false);
+                    as.setDateModified(Instant.now());
+                    as.getSubmitters().stream().filter(sb -> sb.getSubmitter().equals(submitterId)).findFirst().ifPresent(sb -> sb.setSubmittee(false));
```

Comment:
use findAny vs findFirst

Reply:
findFirst() has been changed to findAny() in every place where it appeared.